test/test.js

```diff
@@ -4909,6 +4909,102 @@ exports['tap - doto alias'] = function (test) {
     test.done();
 };
 
+
+exports.flatTap = {
+    'flatTap - noValueOnError': noValueOnErrorTest(_.flatTap(function (x) { return _(); })),
+    'flatTap - returnsSameStream': returnsSameStreamTest(function (s) {
+        return s.flatTap(function (x) { return _([2]); });
+    }, [1], [1]),
+    'flatTap - On a empty stream': function (test) {
+        test.expect(1);
+        var s = _([]).flatTap(function (x) {
+            return _([6]);
+        });
+
+        s.pull(valueEquals(test, _.nil));
+        test.done();
+    },
+    'flatTap - Returns an Empty stream': function (test) {
+        test.expect(1);
+        var s = _([3]).flatTap(function (x) {
+            return _([]);
+        });
+
+        s.pull(valueEquals(test, 3));
+        test.done();
+    },
+    'flatTap - Emits an multiple values': function (test) {
+        test.expect(1);
+        var s = _([3]).flatTap(function (x) {
+            return _([5, 6, 7, 8]);
+        });
+
+        s.pull(valueEquals(test, 3));
+        test.done();
+    },
+    'flatTap - argument function throws': function (test) {
```

Comment:
Can you add a similar test to this, but for the stream (i.e., the return value of the argument function) emitting an error instead.

Something like this,

```javascript
_([1, 2, 3, 4]).flatTap((x) => _((push) => {
    push(err);
    push(null, _.nil);
}));
````